### ISSUE TITLE
Refactor custom `available_to_order?` logic

### ISSUE THREAD

**Issue by sauloperez** [openfoodfoundation/openfoodnetwork#2205]
## Description

It turns out that `ShippingMethod#available_to_order?` is gone in Spree 2.0 and we rely on it to add our custom logic.

### Current status 

On top of the Spree's checks to know whether a shipping method is eligible to be used for an order, we also check that the order's distributor is one of the enterprises that can use the shipping method.

In other words, while Spree defined it like:

```ruby
def available_to_order?(order)
  available?(order) &&
  within_zone?(order) &&
  category_match?(order) &&
  currency_match?(order)
end
```

We extended it with:

https://github.com/openfoodfoundation/openfoodnetwork/blob/236ba9328d3acc99e26f265f97441439149d3252/app/models/spree/shipping_method_decorator.rb#L45-L49

### How does it look in Spree 2.0?

In https://github.com/spree/spree/commit/e1ca57421ce2efb8ee4715bda448158af5b7e90f `available_to_order?`'s logic is moved to [Spree::StockEstimator#shipping_methods(package)](https://github.com/spree/spree/commit/e1ca57421ce2efb8ee4715bda448158af5b7e90f#diff-c917ee5e7ac233c8f9f9d36c44854e15R29), so we need that method to also take our logic into account.

As long as we have investigated some needs have been identified:

* We'll need to a use a default `ShippingCategory` at instance level since [shipping methods are now fetched through the categories](https://github.com/spree/spree/blob/2-0-stable/core/app/models/spree/stock/package.rb#L84-L86) and [products now must specify a shipping category](https://github.com/spree/spree/blob/2-0-stable/core/app/models/spree/product.rb#L72). So far, we haven't used categories at all because [Spree allowed them to be nil](https://github.com/spree/spree/blob/1-3-stable/core/app/models/spree/shipping_method.rb#L44).

**Comment by HugsDaniel** [comment]
Somewhat related https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/overrides/spree/admin/shipping_methods/_form/replace_form_fields.html.haml.deface#L37

**Comment by enricostano** [comment]
Also maybe related https://github.com/openfoodfoundation/openfoodnetwork/issues/2126

**Comment by RachL** [comment]
Also maybe https://github.com/openfoodfoundation/openfoodnetwork/pull/2039#issuecomment-375137267

**Comment by sauloperez** [comment]
And also likely related to https://github.com/openfoodfoundation/openfoodnetwork/pull/1627

**Comment by sauloperez** [comment]
Closed by #2256
